### PR TITLE
Always run `tail_expr_drop_order` lint in promoted MIR query

### PR DESCRIPTION
**Pull request by compiler-errors**
Rather than running the lint at the beginning of `mir_drops_elaborated_and_const_checked`, run it at the end of `mir_promoted`.  This should ensure that the lint gets picked up when running `cargo fix`, which runs with `--emit metadata` and therefore doesn't actually query `mir_drops_elaborated_and_const_checked`.

We could probably push this down into `mir_built` too? but I don't really see a good reason to do so.

https://github.com/rust-lang/rust/issues/132861#issuecomment-2539960199

cc @ehuss

### CODE REVIEW

File: tests/ui/drop/lint-tail-expr-drop-order.rs [hunk 1]
```diff
@@ -2,7 +2,6 @@
 // This lint is to capture potential change in program semantics
 // due to implementation of RFC 3606 <https://github.com/rust-lang/rfcs/pull/3606>
 //@ edition: 2021
-//@ build-fail
```
Comment (compiler-errors) [inline tests/ui/drop/lint-tail-expr-drop-order.rs:5]: A `build-fail` became a `check-fail`, which is evidence enough of this being a sufficient fix.

File: tests/ui/error-codes/E0452.rs [hunk 2]
```diff
@@ -2,5 +2,7 @@
                     //~| ERROR E0452
                     //~| ERROR E0452
                     //~| ERROR E0452
+                    //~| ERROR E0452
```
Comment (compiler-errors) [inline tests/ui/error-codes/E0452.rs:5]: Duplicated attr errors is due to a call to `lints_that_dont_need_to_run` in check-builds. This is deduplicated to the user, so it doesn't matter.

**Comment by lqd** [comment]
bors doesn’t seem to like a couple of errs PRs smh

@bors ping

**Review by oli-obk — APPROVED** [review]

**Comment by lqd** [comment]
Reopening to try and fix bors

**Comment by Kobzol** [comment]
Homu probably got wind of our attempts to get rid of him.

**Comment by compiler-errors** [comment]
I don't believe this is significant enough to warrant a fix.

**Pull request merged**